src/ripple/app/ledger/PendingSaves.h

```diff
@@ -0,0 +1,86 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2012-2015 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_APP_PENDINGSAVES_H_INCLUDED
+#define RIPPLE_APP_PENDINGSAVES_H_INCLUDED
+
+#include <ripple/protocol/Protocol.h>
+#include <boost/container/flat_set.hpp>
+#include <mutex>
+
+namespace ripple {
+
+/** Keeps track of which ledgers haven't been fully saved.
+
+    During the ledger building process this collection will keep
+    track of those ledgers that are being built but have not yet
+    been completely written.
+*/
+class PendingSaves
+{
```

Comment:
You could make this a generic thread-safe `flat_set` templated on element type and then type alias `PendingSaves`.